subcommand/injector/command.go

```diff
@@ -353,47 +358,139 @@ func (c *Command) certWatcher(ctx context.Context, ch <-chan cert.Bundle, client
 			}
 		}
 
-		// If there is a MWC name set, then update the CA bundle.
+		// If there is an MWC name set, then update the CA bundle.
 		if isLeader && c.flagAutoName != "" && len(bundle.CACert) > 0 {
-			// The CA Bundle value must be base64 encoded
-			value := base64.StdEncoding.EncodeToString(bundle.CACert)
-			payload := []byte(fmt.Sprintf(
-				`[{
-					"op": "add",
-					"path": "/webhooks/0/clientConfig/caBundle",
-					"value": %q
-				}]`, value))
-
-			var err error
-			switch adminAPIVersion {
-			// AdmissionregistrationV1 is valid in k8s 1.16+
-			case "v1":
-				_, err = clientset.AdmissionregistrationV1().
-					MutatingWebhookConfigurations().
-					Patch(ctx, c.flagAutoName, types.JSONPatchType, payload,
-						metav1.PatchOptions{})
-			// AdmissionregistrationV1beta1 is present in k8s 1.9 - 1.19
-			case "v1beta1":
-				_, err = clientset.AdmissionregistrationV1beta1().
-					MutatingWebhookConfigurations().
-					Patch(ctx, c.flagAutoName, types.JSONPatchType, payload,
-						metav1.PatchOptions{})
-			default:
-				err = fmt.Errorf("unknown Admissionregistration API version %q", adminAPIVersion)
-			}
+			// Check the current bundles and only update if necessary.
+			currentBundles, err := c.getCurrentCABundles(ctx, clientset, adminAPIVersion)
 			if err != nil {
 				c.UI.Error(fmt.Sprintf(
-					"Error updating MutatingWebhookConfiguration: %s",
+					"Error getting MutatingWebhookConfiguration: %s",
 					err))
 				continue
 			}
+			if len(currentBundles) == 0 || !bytes.Equal(currentBundles[0], bundle.CACert) {
+				err = c.updateCABundle(ctx, &bundle, clientset, adminAPIVersion)
+				if err != nil {
+					c.UI.Error(fmt.Sprintf(
+						"Error updating MutatingWebhookConfiguration: %s",
+						err))
+					continue
+				}
+			}
 		}
 
 		// Update the certificate
 		c.cert.Store(&crt)
 	}
 }
 
+func watchWebhooks(ctx context.Context, clientset *kubernetes.Clientset, adminAPIVersion string) (watch.Interface, error) {
+	switch adminAPIVersion {
+	// AdmissionregistrationV1 is valid in k8s 1.16+
+	case "v1":
+		return clientset.AdmissionregistrationV1().MutatingWebhookConfigurations().Watch(ctx, metav1.ListOptions{})
```

Comment:
This will return watch events for all mutating webhook configs on the system? Are there any ListOptions that could filter that list so it isn't triggered on non-vault-k8s webhook changes?

I'm also wondering if you considered setting up an informer here. IIRC a straight `Watch` isn't as robust when it comes to network interruptions (these discussions are a little old, so maybe my knowledge is outdated): 
* https://github.com/kubernetes-client/javascript/issues/377
* https://stackoverflow.com/questions/59544139/kubernetes-client-go-watch-interface-vs-cache-newinformer-vs-cache-newsharedi

Another option would be to poll against the informer cache's lister interface, as we do with the cert secret: https://github.com/hashicorp/vault-k8s/blob/0c69acbf33605f66cd1316fd9cd5380e7560f84d/helper/cert/source_gen.go#L263

Reply:
Ah, perfect. I didn't know about informers. That sounds much more robust.